syntax = "proto3";
package s2s;

message NodeRegister
{
    message Request
    {
        int32 AreaId = 1;   //区服id
        int32 NodeId = 2;   //节点id
        string Address = 3; //地址
        string ServerName = 4;  //服务器名字
        repeated string Services = 5;   //服务名列表
    }
}

message NodeData
{
    message NodeInfo
    {
        int32 AreaId = 1;   //区服id
        int32 NodeId = 2;   //节点id
        string Address = 3; //地址
        string ServerName = 4;  //服务器名字
        repeated string Services = 5;   //服务名列表
    }
    message Request
    {
        int32 AreaId = 1; //区服id
    }
    message Response
    {
        repeated NodeInfo NodeArray = 1;   //节点列表
    }
}